src/airport.js

```diff
@@ -1 +1,98 @@
+class Airport {
 
+    constructor(airportID) {
+        this.airportID = airportID;
+        this.listOfPlanesAtAirport = [];
+        this.airportMaxCapacity = 2;
+        this.weatherSafeForAction = true;
+    }
+
+    get getAirportID() {
+        return this.airportID;
+    }
+
+    get getListOfPlanesAtAirport() {
+        return this.listOfPlanesAtAirport;
+    }
+
+    get getAirportMaxCapacity() {
+        return this.airportMaxCapacity;
+    }
+
+    setAirportMaxCapacity(newCapacity) {
+        if (newCapacity >= this.numberOfPlanesAtAirport() && typeof newCapacity === 'number' && newCapacity >= 0) {
+            this.airportMaxCapacity = newCapacity
+        }
+    }
+
+
+    instructToLand(plane) {
+        if (this.preLandChecks(plane)) {
+            this.listOfPlanesAtAirport = [...this.listOfPlanesAtAirport, plane];
+            plane.setInFlight(false);
+        }
+    }
+
+    isAirportFull() {
+        return this.airportMaxCapacity === this.getListOfPlanesAtAirport.length;
+    }
+
+    instructToTakeOff2(plane) {
+
+        if (this.preTakeOffChecks(plane)) {
+
+            this.listOfPlanesAtAirport = this.listOfPlanesAtAirport.filter(findPlanes);
```

Comment:
Think this reverse use of making an array full of plane objects that didn't meet the criteria using filer() was a  good approach